lib/optimize/UglifyJsPlugin.js

```diff
@@ -102,6 +103,54 @@ class UglifyJsPlugin {
 						for(let k in options.output) {
 							output[k] = options.output[k];
 						}
+						const extractedComments = [];
+						if(options.extractComments) {
+							const condition = {};
+							if(typeof options.extractComments === "string" || options.extractComments instanceof RegExp) {
+								// extractComments specifies the extract condition and output.comments specifies the preserve condition
+								condition.preserve = output.comments;
+								condition.extract = options.extractComments;
+							} else if(Object.prototype.hasOwnProperty.call(options.extractComments, "condition")) {
+								// Extract condition is given in extractComments.condition
+								condition.preserve = output.comments;
+								condition.extract = options.extractComments.condition;
+							} else {
+								// No extract condition is given. Extract comments that match output.comments instead of preserving them
+								condition.preserve = false;
+								condition.extract = output.comments;
+							}
+
+							// Ensure that both conditions are functions
+							["preserve", "extract"].forEach(key => {
+								switch(typeof condition[key]) {
+									case "boolean":
+										var b = condition[key];
+										condition[key] = () => b;
+									case "function": // eslint-disable-line no-fallthrough
+										break;
+									case "string":
+										if(condition[key] === "all") {
+											condition[key] = () => true;
+											break;
+										}
+										condition[key] = new RegExp(condition[key]);
+									default: // eslint-disable-line no-fallthrough
```

Comment:
Maybe solve this without fallthrough. It requires an additional line but is more readable..